src/Auth/BaseAuthenticate.php

```diff
@@ -153,6 +153,10 @@ protected function _query($username)
             $finder = key($finder);
         }
 
+        if (empty($options['username'])) {
```

Comment:
You might want to use isset(). This method will forbid the use of 0 as a username. It sounds silly but someone will eventually do it.